lib/paper_trail/frameworks/rails/controller.rb

```diff
@@ -61,14 +61,23 @@ def set_paper_trail_enabled_for_controller
       end
 
       # Tells PaperTrail who is responsible for any changes that occur.
-      def set_paper_trail_whodunnit
-        ::PaperTrail.whodunnit = user_for_paper_trail if ::PaperTrail.enabled_for_controller?
+      def set_paper_trail_whodunnit(&block)
+        if ::PaperTrail.enabled_for_controller?
+          ::PaperTrail.with_whodunnit(user_for_paper_trail, &block)
+        else
+          yield
+        end
       end
 
       # Tells PaperTrail any information from the controller you want to store
       # alongside any changes that occur.
       def set_paper_trail_controller_info
```

Comment:
To fix the failing tests, 

```
NameError: undefined local variable or method `block'
```

maybe you need to add `&block` to the arguments here?